src/util/earthEngine.js

```diff
@@ -2,11 +2,23 @@ import i18n from '@dhis2/d2-i18n'
 import { loadEarthEngineWorker } from '../components/map/MapApi.js'
 import { apiFetch } from './api.js'
 import { formatStartEndDate } from './time.js'
+import { EE_MONTHLY } from '../constants/periods.js'
+import { legacyNighttimeDatasetId } from '../constants/earthEngineLayers/legacy/nighttime_DMSP-OLS.js'
 
 export const classAggregation = ['percentage', 'hectares', 'acres']
 
 export const hasClasses = (type) => classAggregation.includes(type)
 
+// Translates from dynamic to static filters
+export const translateFilters = (filters, ...args) =>
```

Comment:
A function like this should really have some unit tests. Try asking gh copilot to write some! You might have to make some adjustments, but it usually provides a really good start.

Also, what about just naming the function according to what it does, e.g.
getStaticFiltersFromDynamic
convertFiltersFromDynamicToStatic

Reply:
Will fix!

Reply:
Changed to more descriptive names:
- getStaticFilterFromPeriod
- getStaticFiltersFromDynamic

Added some unit tests for the `getStaticFilterFromPeriod` function (which also includes `getStaticFiltersFromDynamic`)